docs/src/model_setup/boundary_conditions.md

```diff
@@ -1,62 +1,173 @@
+2. [`ValueBoundaryCondition`](@ref) (Dirchlet) specifies the value of a field on
+   the given boundary, which when used in combination with a turbulence closure
+   results in a flux across the boundary. For example, `ValueBoundaryCondition(0)` is used
+   to specify no-slip boundary conditions on velocity components tangential to the given boundary.
 
-1. Boundary conditions depend on the grid topology and can only be non-default or non-`Periodic` in `Bounded` directions.
-   Tracer boundary conditions are no flux by default in `Bounded` directions.
-   Momentum boundary conditions are free-slip for tangential components and impenetrable for wall-normal components in `Bounded` directions.
-   
-2. Another way to say point 1 is that you'll never need to set:
-    * `Periodic` boundary conditions (default for `Periodic` directions);
-    * Impenetrable / "no normal flow" boundary conditions (default for wall-normal momentum components in `Bounded` directions);
-    * "No flux" or "free slip" boundary conditions (default for tracers and wall-tangential momentum components in `Bounded` directions).
+   Examples with `ValueBoundaryCondition`:
 
-3. `ValueBoundaryCondition` (aka "Dirichlet" boundary conditions) models boundary fluxes given a field's diffusive flux model, and assuming that a field has the prescribed value on the boundary.
-   _Note_: You cannot use `ValueBoundaryCondition` on a wall-normal velocity component; you must use `Open` for that.
-   Examples where you might use `ValueBoundaryCondition`:
    * Prescribe a surface to have a constant temperature, like 20 degrees.
      Heat will then flux in and out of the domain depending on the temperature difference between the surface and the interior, and the temperature diffusivity.
    * Prescribe a velocity tangent to a boundary as in a driven-cavity flow (for example), where the top boundary is moving.
      Momentum will flux into the domain do the difference between the top boundary velocity and the interior velocity, and the prescribed viscosity.
 
-4. `FluxBoundaryCondition` _directly_ prescribes the flux of a quantity across a boundary rather than calculating it given a viscosity or diffusivity.
-   For example, sunlight absorbed at the ocean surface imparts a temperature flux that heats near-surface fluid.
-   If there is a known `diffusivity`, you can express `FluxBoundaryCondition(flux)` using `GradientBoundaryCondition(-flux / diffusivity)` (aka "Neumann" boundary condition).
-   But when `diffusivity` is not known or is variable (as for large eddy simulation, for example), it's convenient and more straightforward to apply `FluxBoundaryCondition`.
+   _Note_: Do not use `ValueBoundaryCondition` on a wall-normal velocity component.
+   `ImpenetrableBoundaryCondition` is internally enforced and thus only needs to be specified
+   for "additional" fields created outside model constructors.
```

Comment:
I think statement may be a bit confusing. I myself can't think of a situation where the avg user needs to know/use this. So maybe if there's no such situation we could remove the "additional fields" comments? If there are such situations, maybe mention a quick example? (No code necessary though)

Reply:
We've seen this error numerous times; that's why I'm mentioning it.

Reply:
Hm, I wasn't sure what you meant by "this". We often see the error that people attempt to enforce impenetrability with `ValueBoundaryCondition(0)` so I'm trying to be proactive in preventing it (and I see why one might make that mistake).

As for using impenetrable boundary condition, it comes up when creating slip velocity fields that are used with `AdvectiveForcing`.

Reply:
I noticed that we include a more detailed statement about `AdvectiveForcing` just below so I simplified this line.

Reply:
Sorry, just to clarify. This part

```
   _Note_: Do not use `ValueBoundaryCondition` on a wall-normal velocity component.
   `ImpenetrableBoundaryCondition` is internally enforced
```

Is important and I agree that it should be there.

This part:

```
only needs to be specified for "additional" fields created outside model constructors.
``` 
is the one that I think may be confusing without a quick mention of an example. It confused me, and I'm a regular contributor. But maybe I just don't know enough about field BCs...

Reply:
I think if its confusing you should ignore it (it's only relevant to people building `AdvectiveForcing`). How do we make that clear?